baselines/pfsp/pfsp_gpu_cuda.c

```diff
@@ -0,0 +1,570 @@
+
+        pushBack(pool, child);
+        *tree_loc += 1;
+      }
+    }
+  }
+
+  free(lb_begin);
+}
+
+void decompose_lb2(const int jobs, const lb1_bound_data* const lbound1, const lb2_bound_data* const lbound2,
+		   const Node parent, int* best, unsigned long long int* tree_loc, unsigned long long int* num_sol,
+		   SinglePool* pool)
+{
+  for (int i = parent.limit1+1; i < jobs; i++) {
+    Node child;
+    memcpy(child.prmu, parent.prmu, jobs * sizeof(int));
+    swap(&child.prmu[parent.depth], &child.prmu[i]);
+    child.depth = parent.depth + 1;
+    child.limit1 = parent.limit1 + 1;
+
+    int lowerbound = lb2_bound(lbound1, lbound2, child.prmu, child.limit1, jobs, *best);
+
+    if (child.depth == jobs) { // if child leaf
+      *num_sol += 1;
+
+      if (lowerbound < *best) { // if child feasible
+        *best = lowerbound;
+      }
+    } else { // if not leaf
+      if (lowerbound < *best) { // if child feasible
+        pushBack(pool, child);
+        *tree_loc += 1;
+      }
+    }
+  }
+}
+
+void decompose(const int jobs, const int lb, int* best,
+	       const lb1_bound_data* const lbound1, const lb2_bound_data* const lbound2, const Node parent,
+	       unsigned long long int* tree_loc, unsigned long long int* num_sol, SinglePool* pool)
+{
+  switch (lb) {
+  case 0: // lb1_d
+    decompose_lb1_d(jobs, lbound1, parent, best, tree_loc, num_sol, pool);
+    break;
+
+  case 1: // lb1
+    decompose_lb1(jobs, lbound1, parent, best, tree_loc, num_sol, pool);
+    break;
+
+  case 2: // lb2
+    decompose_lb2(jobs, lbound1, lbound2, parent, best, tree_loc, num_sol, pool);
+    break;
+  }
+}
+
+// Generate children nodes (evaluated on GPU) on CPU
+void generate_children(Node* parents, const int size, const int jobs, int* bounds,
+		       unsigned long long int* exploredTree, unsigned long long int* exploredSol, int* best, SinglePool* pool)
+{
+  for (int i = 0; i < size; i++) {
+    Node parent = parents[i];
+    const uint8_t depth = parent.depth;
+
+    for (int j = parent.limit1+1; j < jobs; j++) {
+      const int lowerbound = bounds[j + i * jobs];
+
+      // If child leaf
+      if(depth + 1 == jobs){
+	*exploredSol += 1;
+
+	// If child feasible
+	if(lowerbound < *best) *best = lowerbound;
+
+      } else { // If not leaf
+	if(lowerbound < *best) {
+	  Node child;
+	  memcpy(child.prmu, parent.prmu, jobs * sizeof(int));
+	  swap(&child.prmu[parent.depth], &child.prmu[j]);
+	  child.depth = parent.depth + 1;
+	  child.limit1 = parent.limit1 + 1;
+	  
+	  pushBack(pool, child);
+	  *exploredTree += 1;
+	}
+      }
+    }
+  }
+}
+
+// Single-GPU PFSP search
+void pfsp_search(const int inst, const int lb, const int m, const int M, int* best,
+		 unsigned long long int* exploredTree, unsigned long long int* exploredSol,
+		 double* elapsedTime)
+{
+
+  // Initializing problem
+  int jobs = taillard_get_nb_jobs(inst);
+  int machines = taillard_get_nb_machines(inst);
+  int count = 0;
```

Comment:
c'est tjrs nécessaire de compter les itérations du while ? Si non, supprimer `count` partout.